src/scipp/interpolate/__init__.py

```diff
@@ -15,7 +17,20 @@ def _midpoints(var, dim):
     return a + 0.5 * (b - a)
 
 
-@wrap1d(is_partial=True)
+def _drop_masked(da, dim):
+    mask = irreducible_mask(da.masks, dim)
+    if mask is not None:
+        da = da.copy(deep=False)
+        name = uuid.uuid4().hex
+        da.coords[name] = mask
+        da = da.groupby(name).copy(0)
+        for name in list(da.masks):
```

Comment:
Why convert to `list` here?

Reply:
I think you can't remove from a dict while iterating it?